src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -229,7 +247,11 @@ rpmostreed_deployment_generate_variant (OstreeSysroot *sysroot,
       variant_add_commit_details (&dict, "base-", base_commit);
     }
   else
-    base_checksum = g_strdup (csum);
+    {
+      base_checksum = g_strdup (csum);
+      /* we would not consider endoflife attribute for layered commits */
```

Comment:
hmm, for that case I agree, your way makes more sense overall :), will work on to change it.